spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RibbonLoadBalancingHttpClient.java

```diff
@@ -64,9 +62,28 @@ public RibbonLoadBalancingHttpClient(final ILoadBalancer lb) {
 	}
 
 	@Override
+	public void initWithNiwsConfig(IClientConfig clientConfig) {
```

Comment:
Yes it is called because `RibbonLoadBalancingHttpClient`'s superclass is `IClientConfigAware`. When `SpringClientFactory` is used to get the instance it [calls `instantiateWithConfig`](https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/SpringClientFactory.java#L186) to create the instance the first time, and `instanciateWithConfig` [calls  the object's `initWithNiwsConfig`](https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/SpringClientFactory.java#L157) if it is `IClientConfigAware`.

`HttpClientRibbonCommandFactory` uses a `SpringClientFactory` to get the `RibbonLoadBalancingHttpClient` instance, hence `initWithNiwsConfig` is called. This is why a removed the unnecessary `client.setClientConfig(this.clientFactory.getClientConfig(serviceId));` there.

I've updated the tests to use a `SpringClientFactory` instead of calling directly `initWithNiwsConfig` to make it clear it's working.

I've also tested and seen the expected behaviour in my own application adding the following property in my boostrap.yml : 
```
ribbon:
  FollowRedirects: true/false
```